staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_apis.go

```diff
@@ -39,6 +39,13 @@ type apisHandler struct {
 	codecs serializer.CodecFactory
 	lister listers.APIServiceLister
 	mapper request.RequestContextMapper
+
+	// This is the list of API services of the deligated server. It may take
+	// some time for them to get registered as internal with aggregator but
+	// we can return them as part of API discovery anyway as they are served
+	// regardless of their registration state (all unknown API calls will be
+	// sent to delegated server anyway).
+	internalServices []*apiregistrationapi.APIService
```

Comment:
This adds a second source of truth for which API groups are active, does not honor the order specified in the APIService objects, special-cases locally routed services, and assumes that delegated handling will route to unavailable APIService endpoints